File: reni-frontend/src/main/reni-frontend/src/app/modules/intranet/services/recursos-seguridad.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { OutResponse } from '../../sesion/dto/response/out.response';
import { RecursosSeguridadListarCargoResponse } from '../dto/response/recursos-seguridad-listar-cargo.response';
import { RecursosSeguridadListarTipoDocumResponse } from '../dto/response/recursos-seguridad-listar-tipo-docum.response';

@Injectable()
export class RecursosSeguridadService {

  constructor(private http: HttpClient) { }

  public listarTipoDocumento(): Observable<OutResponse<RecursosSeguridadListarTipoDocumResponse[]>> {
    return this.http.post<OutResponse<RecursosSeguridadListarTipoDocumResponse[]>>(`${environment.reniBackendUrl}/recursos-seguridad/listarTipoDocumento`, {});
  }

  public listarCargo(): Observable<OutResponse<RecursosSeguridadListarCargoResponse[]>> {
    return this.http.post<OutResponse<RecursosSeguridadListarCargoResponse[]>>(`${environment.reniBackendUrl}/recursos-seguridad/listarCargo`, {});
  }
}

File: reni-backend/src/main/java/pe/gob/mimp/reni/service/RecursosSeguridadService.java
package pe.gob.mimp.reni.service;

import java.util.List;

import pe.gob.mimp.reni.dto.response.OutResponse;
import pe.gob.mimp.reni.dto.response.RecursosSeguridadListarCargoResponse;
import pe.gob.mimp.reni.dto.response.RecursosSeguridadListarTipoDocumResponse;

public interface RecursosSeguridadService {

	public OutResponse<List<RecursosSeguridadListarTipoDocumResponse>> listarTipoDocumento();
	
	public OutResponse<List<RecursosSeguridadListarCargoResponse>> listarCargo();
}
